src/Core/BufferLine.re

```diff
@@ -88,8 +98,17 @@ module Internal = {
 let make = (~indentation, raw: string) => {
   // Create a cache the size of the string - this would be the max length
   // of the UTF8 string, if it was all 1-byte unicode characters (ie, an ASCII string).
-  let characters = emptyArray;
-  {indentation, raw, characters, nextByte: 0, nextIndex: 0, nextPosition: 0};
+  let characters = emptyArrayCharacters;
+  let byteMap = emptyArrayIndex;
```

Comment:
This seems like unnecessary indirection when they can just be assigned to the record fields directly.